nav2_system_tests/src/gps_navigation/tester.py

```diff
@@ -0,0 +1,222 @@
+        try:
+            rclpy.spin_until_future_complete(self, get_result_future)
+            status = get_result_future.result().status
+            result = get_result_future.result().result
+            self.action_result = result
+        except Exception as e:  # noqa: B902
+            self.error_msg(f"Service call failed {e!r}")
+
+        if status != GoalStatus.STATUS_SUCCEEDED:
+            self.info_msg(f"Goal failed with status code: {status}")
+            return False
+        if len(result.missed_waypoints) > 0:
+            self.info_msg(
+                "Goal failed to process all waypoints,"
+                " missed {0} wps.".format(len(result.missed_waypoints))
+            )
+            return False
+
+        self.info_msg("Goal succeeded!")
+        return True
+
+    def setStopFailureParam(self, value):
+        req = SetParameters.Request()
+        req.parameters = [Parameter('stop_on_failure',
+                                    Parameter.Type.BOOL, value).to_parameter_msg()]
+        future = self.param_cli.call_async(req)
+        rclpy.spin_until_future_complete(self, future)
+
+    def shutdown(self):
+        self.info_msg("Shutting down")
+
+        self.action_client.destroy()
+        self.info_msg("Destroyed follow_gps_waypoints action client")
+
+        transition_service = "lifecycle_manager_navigation/manage_nodes"
+        mgr_client = self.create_client(
+            ManageLifecycleNodes, transition_service)
+        while not mgr_client.wait_for_service(timeout_sec=1.0):
+            self.info_msg(
+                f"{transition_service} service not available, waiting...")
+
+        req = ManageLifecycleNodes.Request()
+        req.command = ManageLifecycleNodes.Request().SHUTDOWN
+        future = mgr_client.call_async(req)
+        try:
+            rclpy.spin_until_future_complete(self, future)
+            future.result()
+        except Exception as e:  # noqa: B902
+            self.error_msg(f"{transition_service} service call failed {e!r}")
+
+        self.info_msg(f"{transition_service} finished")
+
+    def cancel_goal(self):
+        cancel_future = self.goal_handle.cancel_goal_async()
+        rclpy.spin_until_future_complete(self, cancel_future)
+
+    def info_msg(self, msg: str):
+        self.get_logger().info(msg)
+
+    def warn_msg(self, msg: str):
+        self.get_logger().warn(msg)
+
+    def error_msg(self, msg: str):
+        self.get_logger().error(msg)
+
+
+def main(argv=sys.argv[1:]):
+    rclpy.init()
+
+    # wait a few seconds to make sure entire stacks are up
+    time.sleep(10)
+
+    wps = [[55.944831, -3.186998], [55.944818, -3.187075], [55.944782, -3.187060]]
+
+    test = GpsWaypointFollowerTest()
+    test.setWaypoints(wps)
+
+    # wait for poseCallback
+
+    result = test.run(True, False)
+    assert result
+
+    # preempt with new point
+    test.setWaypoints([wps[0]])
+    result = test.run(False, False)
```

Comment:
You never check `result` if it failed or succeeded

Reply:
For several of them throughout this file, please check and make sure we validate each test

Reply:
The tests on the GPS WPF were almost a direct copy paste from the [existing tests for the cartesian wpf](https://github.com/ros-planning/navigation2/blob/main/nav2_system_tests/src/waypoint_follower/tester.py). Some results are not checked there as well I think because they dont necessarily return whether the action succeeded, but just wheteher the action was accepted. This could still be asserted though, so let me know if I should change this in the gps wpf tests and the cartesian wpf tests as well.